### ISSUE TITLE
vars_prompt: allow variables substitution and conditionals

### ISSUE THREAD

**Issue by mathieumd** [ansible/ansible#13263]
##### Issue Type: Feature Idea

Related to  #11239.
##### Ansible Version:

```
ansible 2.0.0 (devel 9b9fb51d9d) last updated 2015/11/18 11:51:54 (GMT +200)
  lib/ansible/modules/core: (detached HEAD b1c0249045) last updated 2015/11/18 11:51:55 (GMT +200)
  lib/ansible/modules/extras: (detached HEAD 7da1f8d4ca) last updated 2015/11/18 11:51:55 (GMT +200)
  config file = /home/.../ansible.cfg
  configured module search path = Default w/o overrides
```
##### Environment:

running Ansible from: N/A
what OS are you managing?: N/A
##### Summary:

It seems `vars_prompt` does not allow:
- **variable substitution**: `default: "{{ ansible_default_ipv4.interface }}"` is displayed as-is: "`My prompt [{{ ansible_default_ipv4.interface }}]:`"
- **conditional features**: Even with a `when: my_var is undefined` to a `vars_prompt` bloc, and with the command-line `ansible-playbook ... -e "my_var=test"`, it still prompt for the value.
##### Steps To Reproduce:

Add a `vars_prompt` to your playbook:

``` yaml
vars_prompt:
- name: my_var
  prompt: "My var"
  default: "{{ ansible_default_ipv4.interface }}"
  private: no
  when: my_var is undefined
```

And call it by setting the variable `my_var`:

``` bash
ansible-playbook test.yml -e "my_var=test" -v
```
##### Expected Results:

I should not be prompted for `my_var` value.
##### Actual Results:

I am prompted for `my_var`, and default suggested value is "`{{ ansible_default_ipv4.interface }}`" as-is.

[This issue has been initially erroneously posted on [ansible/ansible-modules-core](https://github.com/ansible/ansible-modules-core/issues/2517)]

**Comment by bcoca** [comment]
i would expect that templated values for facts are not substituted, as vars_prompt executes before we start looping over hosts. Even if it didn't this would be an issue, if `hosts: group_with_10_hosts`, which IP would you expect displayed?

**Comment by mathieumd** [comment]
The IP of the currently processed machine, no?

I was expecting that "`{{ ansible_default_ipv4.interface }}`" would be translated to "`eno16780032`" (by example), providing a dynamic and safe default value. (We use this in a playbook which setup the network of a freshly installed RHEL/CentOS 7, where default interface's name is not always `eth0`, due to [Systemd predictable interface name](http://www.freedesktop.org/wiki/Software/systemd/PredictableNetworkInterfaceNames/).

For the `when:` statement, the idea is to have a playbook prompting for variables values _only_ when those variables are still undefined.

**Comment by bcoca** [comment]
at the time vars_prompt runs only extra vars are defined, so the when will not work with host/group vars nor facts.

and you are expecting the var to be translated to a random fact from a host? that won't happen.

what WILL happen is when you use that value in the play, it WILL translate to any facts gathered, but it is not available at the point of prompting.

You might want to look into the pause module that behaves as a task and has access to the gathered facts and host/group vars.

**Comment by rakshazi** [comment]
Affetcts 2.2

**Comment by ansibot** [comment]
@mathieumd Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibotdev** [comment]
@mathieumd You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
